src/Nancy/Json/Simple/NancySerializationStrategy.cs

```diff
@@ -0,0 +1,217 @@
+﻿namespace Nancy.Json.Simple
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using Nancy.Extensions;
+
+    internal class NancySerializationStrategy : PocoJsonSerializerStrategy
+    {
+        private readonly bool retainCasing;
+
+        private readonly bool useIso8601;
+
+        private readonly List<JavaScriptConverter> converters = new List<JavaScriptConverter>();
+        private readonly List<JavaScriptPrimitiveConverter> primitiveConverters = new List<JavaScriptPrimitiveConverter>();
+        internal static readonly long InitialJavaScriptDateTicks = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc).Ticks;
+        static readonly DateTime MinimumJavaScriptDate = new DateTime(100, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+
+        public NancySerializationStrategy() : this(false) { }
+
+        public NancySerializationStrategy(
+            bool retainCasing, 
+            bool registerConverters = true,
+            bool useIso8601 = true,
+            IEnumerable<JavaScriptConverter> converters = null,
+            IEnumerable<JavaScriptPrimitiveConverter> primitiveConverters = null)
+        {
+            this.retainCasing = retainCasing;
+            this.useIso8601 = useIso8601;
+            if (registerConverters)
+            {
+                this.RegisterConverters(converters, primitiveConverters);
+            }
+        }
+
+        private void RegisterConverters(IEnumerable<JavaScriptConverter> javaScriptConverters, IEnumerable<JavaScriptPrimitiveConverter> javaScriptPrimitiveConverters)
+        {
+            if (javaScriptConverters != null)
+            {
+                this.RegisterConverters(javaScriptConverters);
+            }
+            if (javaScriptPrimitiveConverters != null)
+            {
+                this.RegisterConverters(javaScriptPrimitiveConverters);
+            }
+        }
+
+        public void RegisterConverters(IEnumerable<JavaScriptConverter> javaScriptConverters)
+        {
+            this.converters.AddRange(javaScriptConverters);
+        }
+
+        public void RegisterConverters(IEnumerable<JavaScriptPrimitiveConverter> javaScriptPrimitiveConverters)
+        {
+            this.primitiveConverters.AddRange(javaScriptPrimitiveConverters);
+        }
+
+        protected override string MapClrMemberNameToJsonFieldName(string clrPropertyName)
+        {
+            return this.retainCasing ? base.MapClrMemberNameToJsonFieldName(clrPropertyName) :
+                clrPropertyName.ToCamelCase();
+        }
+
+        public override object DeserializeObject(object value, Type type)
+        {
+            if (type.IsEnum || (ReflectionUtils.IsNullableType(type) && Nullable.GetUnderlyingType(type).IsEnum))									
+                return value == null ? null : Enum.Parse(ReflectionUtils.IsNullableType(type) ? Nullable.GetUnderlyingType(type) : type, value.ToString(), true);

```

Comment:
- Missing curly braces
- The double ternary operator usage is messy, please refactor